SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+	{
+		if (SchemaLine.Contains(TEXT("\tid =")))
+		{
+			TArray<FString> OutArray;
+			SchemaLine.ParseIntoArrayWS(OutArray);
+			FString ParsedId = OutArray.Last().LeftChop(1);
+
+			if (ParsedId.IsNumeric())
+			{
+				ParsedNamesAndIds.Ids.Push(FCString::Atoi(*ParsedId));
+			}
+		}
+		else if (SchemaLine.Contains(TEXT("component ")))
+		{
+			TArray<FString> OutArray;
+			SchemaLine.ParseIntoArrayWS(OutArray);
+			if ((OutArray.Num() > 1) && (OutArray[0].Compare(TEXT("component")) == 0))
+			{
+				ParsedNamesAndIds.Names.Push(OutArray[1]);
+			}
+		}
+	}
+
+	return ParsedNamesAndIds;
+}
+
+bool TestEqualDatabaseEntryAndSchemaFile(const UClass* CurrentClass, const FString& SchemaOutputFolder, const USchemaDatabase* SchemaDatabase)
+{
+	NamesAndIds ParsedNamesAndIds = ParseAvailableNamesAndIdsFromSchemaFile(SchemaOutputFolder, CurrentClass);
+
+	if (CurrentClass->IsChildOf<AActor>())
+	{
+		const FActorSchemaData* ActorData = SchemaDatabase->ActorClassPathToSchema.Find(CurrentClass->GetPathName());
+		if (ActorData == nullptr)
+		{
+			return false;
+		}
+		else
+		{
+			if (ParsedNamesAndIds.Names.Num() != 1)
+			{
+				return false;
+			}
+
+			if (ActorData->GeneratedSchemaName.Compare(ParsedNamesAndIds.Names[0]) != 0)
+			{
+				return false;
+			}
+
+			// TODO(Alex): how to test this?
+			//ActorData->SubobjectData;
+
+			for (int i = 0; i < ParsedNamesAndIds.Ids.Num(); ++i)
+			{
+				if (ActorData->SchemaComponents[i] != ParsedNamesAndIds.Ids[i])
+				{
+					return false;
+				}
+			}
+		}
+	}
+	else
+	{
+		const FSubobjectSchemaData* SubobjectSchemaData = SchemaDatabase->SubobjectClassPathToSchema.Find(CurrentClass->GetPathName());
+		if (SubobjectSchemaData == nullptr)
+		{
+			return false;
+		}
+		else
+		{
+			if (ParsedNamesAndIds.Names.Num() != ParsedNamesAndIds.Ids.Num())
+			{
+				return false;
+			}
+
+			for (int i = 0; i < ParsedNamesAndIds.Ids.Num(); ++i)
+			{
+				if (SubobjectSchemaData->DynamicSubobjectComponents[i].SchemaComponents[0] != ParsedNamesAndIds.Ids[i])
+				{
+					return false;
+				}
+
+				FString ExpectedComponentName = SubobjectSchemaData->GeneratedSchemaName;
+				ExpectedComponentName += TEXT("Dynamic");
+				ExpectedComponentName.AppendInt(i + 1);
+				if (ParsedNamesAndIds.Names[i].Compare(ExpectedComponentName) != 0)
+				{
+					return false;
+				}
+			}
+		}
+	}
+
+	return true;
+}
+
+FString LoadSchemaFileForClass(const FString& SchemaOutputFolder, const UClass* CurrentClass)
+{
+	FString SchemaFileFolder = TEXT("");
+
+	if (!CurrentClass->IsChildOf<AActor>())
+	{
+		SchemaFileFolder = TEXT("Subobjects");
+	}
+
+	FString FileContent;
+	FFileHelper::LoadFileToString(FileContent, *FPaths::SetExtension(FPaths::Combine(FPaths::Combine(SchemaOutputFolder, SchemaFileFolder), CurrentClass->GetName()), TEXT(".schema")));
+
+	return FileContent;
+}
+
+TArray<UObject*> AllTestClassesArray =
+{
+	USchemaGenObjectStub::StaticClass(),
+	USpatialTypeObjectStub::StaticClass(),
```

Comment:
Could you maybe add a comment describing `SubobjectSchemaData->DynamicSubobjectComponents[i].SchemaComponents[0]` the [0] specifically.